### DISCUSSION TITLE
"[...] the new ActionScript compiler [...]" (from AIR Release Notes 50.0.0.1) -> ?

### DISCUSSION THREAD

**Discussion by 2jfw** [airsdk/Adobe-Runtime-Support#2144] [Q&A]
Based on the **Release Notes** for AIR 50.0.0.1:

> github-494: Incorporating support for encrypted SWF formats
This is a mechanism to support SWF files where the contents of the file have been encrypted (using a
new tool that will be provided as part of the AIR SDK, ultimately this is planned to be an option for _the
new ActionScript compiler_ as well). Note that the SWF is decrypted in memory and so it may still be
possible for users to access the decrypted SWF data if they can do a memory dump of a process

What exactly is meant with "_the new ActionScript compiler_"?
Is something like **ASC3** upcoming or is it just referring to ASC2 (named as the "new compiler" back then, meanwhile maybe a bit outdated ..?).
Also, wondering if there are any plans to be able to compile Flex with the new compiler(s)?

**Comment by ajwfrost** [comment]
Yes I'm never quite sure what to call it! ASC2 / Falcon2 .. basically the one that we ship in the normal "AIR" packages but not the "AIR for Flex" packages. The actual functionality is all in "compiler.jar". We're looking at incrementally improving it rather than any major new re-write or language updates.

The Flex compiler has some other functionality and capabilities that isn't in this one - not 100% sure why that is, I would have thought there must be a technical reason for it (am not sure that there's a commercial one, seeing as Flex licensing should be compatible..). The Royale compiler has been taken forward somewhat I think, not sure whether that could still be used or whether these sorts of changes could be pulled into it. The Flex compiler does appear to be put together in a very different way though...

**Reply by 2jfw** [reply]
So, any chance for using the same compiler for with and without Flex? Would also avoid people overlaying wrong SDKs etc and it would somehow feel "better" to have the new functions/features available for Flex app, too..

**Reply by ajwfrost** [reply]
Yes I'm not actually sure how much effort it would be to pull in the Flex-specific capabilities into the compiler that we have here. A lot of the options were marked 'flex-specific' and are just stubs .. I can ask someone to look into it; we're already looking at a way to make it easier for overlaying the right SDKs/components though..

**Reply by 2jfw** [reply]
Sounds good!
Would be great if there was any chance to either get the new features/functions for the FlexCompiler or -better- maybe even getting rid of it in favor for ACS2..
